### PR TITLE
Profile picture feature for users

### PR DESCRIPTION
**Pull request by naveen17797**
screenshot from 2018-05-04 22-09-39
@teryhill done, need some info on how to add this to menu

### CODE REVIEW

File: interface/usergroup/user_info.php [hunk 1]
```diff
@@ -83,5 +122,103 @@ function(data)
 </HTML>
 <?php
   //  da39a3ee5e6b4b0d3255bfef95601890afd80709 == blank
+
+if (isset($_FILES)) {
+  //images will be saved with their uid
+  $uid = $_SESSION['authId'];
+    //MAKE THE UPLOAD DIRECTORY IF IT DOESN'T EXIST
+  if (realpath("../../profile_pictures/")) {
```
Comment (aethelwulffe) [inline interface/usergroup/user_info.php]: 755 permission if linux...

File: interface/usergroup/user_info.php [hunk 2]
```diff
@@ -83,5 +122,103 @@ function(data)
...
+  //extension check done.
+  //if any validation needed be added, please add it below.
+  $bool = 0;
+  $target_file =  basename($_FILES["profile_picture"]["name"]);
+  $imageFileType = strtolower(pathinfo($target_file,PATHINFO_EXTENSION));
+  $verify_image = getimagesize($_FILES["profile_picture"]["tmp_name"]);
+  if($verify_image) {
+    $mime = $verify_image["mime"];
+    $mime_types = array('image/png',
+                            'image/jpeg',
```
Comment (aethelwulffe) [inline interface/usergroup/user_info.php]: Just need to know if it is an image...your code works, but getimagesize returns false for non images, so....
```
if(isset($_POST["submit"])) {
    $check = getimagesize($_FILES["fileToUpload"]["tmp_name"]);
    if($check !== false) {
        echo "File is an image - " . $check["mime"] . ".";
        $uploadOk = 1;
    } else {
        echo "File is not an image.";
        $uploadOk = 0;
    }
```

**Comment by aethelwulffe** [comment]
Could always use this for printing ID cards...

**Comment by teryhill** [comment]
@aethelwulffe have you been looking at MY notes?

File: interface/usergroup/user_info.php (hunk 2 above)
Comment (naveen17797) [reply interface/usergroup/user_info.php]: already i have made the validation, if it is not a image then $bool = 0;

Comment (naveen17797) [reply interface/usergroup/user_info.php]: if $bool=0 then no file upload takes place, and no db records get updated.

File: interface/usergroup/user_info.php (hunk 1 above)
Comment (naveen17797) [reply interface/usergroup/user_info.php]: windows doesnt need permissions it seems, are you working in windows?
i am using ubuntu it tests well here.

**Comment by aethelwulffe** [comment]
I guess I should do a re-test soon.   Anyone else tested this yet (for a review)?

File: interface/usergroup/user_info.php (hunk 1 above)
Comment (aethelwulffe) [reply interface/usergroup/user_info.php]: Good!

File: interface/usergroup/user_info.php (hunk 2 above)
Comment (aethelwulffe) [reply interface/usergroup/user_info.php]: Gotcha

File: sql/2_0_0-to-2_0_1_upgrade.sql [hunk 3]
```diff
@@ -117,3 +117,7 @@ ALTER TABLE `users` DROP `fullscreen_role`;
 DROP TABLE IF EXISTS `menu_trees`;
 
 DROP TABLE IF EXISTS `menu_entries`;
+
+#IfMissingColumn users picture_url
+ALTER TABLE `users` ADD `picture_url` VARCHAR(2000) NOT NULL AFTER `suffix`;
```
Comment (aethelwulffe) [inline sql/2_0_0-to-2_0_1_upgrade.sql]: 2000?  Heh...uh...

**Comment by teryhill** [comment]
Have tested and it works well.

**Pull request merged**